### ISSUE TITLE
mistland error

### ISSUE THREAD

**Issue by maddog3521** [RandyKnapp/ValheimMods#417]
i was just wondering if there is a new update available for the mistlands as i am getting errors just for epicloot but if you are working on it i am sorry for bothering you with this. you have great mods and really enjoy playing them

**Comment by maddog3521** [comment]
just wondering when the r2modman update will be available for the epicloot mod i really like that mod and enjoy it

**Comment by tebbeh-dev** [comment]
Maybe post your error?

**Comment by maddog3521** [comment]
[Info   :   BepInEx] Loading [Extended Item Data Framework 1.0.7]
[Warning:  HarmonyX] AccessTools.DeclaredMethod: Could not find method for type Inventory and name AddItem and parameters (string, int, float, Vector2i, bool, int, int, long, string)
[Error  : Unity Log] ArgumentException: Undefined target method for patch method static bool ExtendedItemDataFramework.Inventory_AddItemFromLoad_Patch::Prefix(Inventory __instance, Boolean& __result, string name, int stack, float durability, Vector2i pos, bool equiped, int quality, int variant, long crafterID, string crafterName)
Stack trace:
HarmonyLib.PatchClassProcessor.PatchWithAttributes (System.Reflection.MethodBase& lastOriginal) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.PatchClassProcessor.Patch () (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
Rethrow as HarmonyException: Patching exception in method null
HarmonyLib.PatchClassProcessor.ReportException (System.Exception exception, System.Reflection.MethodBase original) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.PatchClassProcessor.Patch () (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.Harmony.<PatchAll>b__11_0 (System.Type type) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.CollectionExtensions.Do[T] (System.Collections.Generic.IEnumerable`1[T] sequence, System.Action`1[T] action) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.Harmony.PatchAll (System.Reflection.Assembly assembly) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.Harmony.CreateAndPatchAll (System.Reflection.Assembly assembly, System.String harmonyInstanceId) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
ExtendedItemDataFramework.ExtendedItemDataFramework.Awake () (at <dfd72d765e1645ba9a4759fa398691c4>:0)
UnityEngine.GameObject:AddComponent(Type)
BepInEx.Bootstrap.Chainloader:Start()
UnityEngine.Application:.cctor()

[Info   :   BepInEx] Loading [Epic Loot 0.8.9]
[Error  : Unity Log] AmbiguousMatchException: Ambiguous match found.
Stack trace:
System.RuntimeType.GetMethodImpl (System.String name, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Reflection.CallingConventions callConv, System.Type[] types, System.Reflection.ParameterModifier[] modifiers) (at <695d1cc93cca45069c528c15c9fdd749>:0)
System.Type.GetMethod (System.String name, System.Reflection.BindingFlags bindingAttr) (at <695d1cc93cca45069c528c15c9fdd749>:0)
HarmonyLib.AccessTools.DeclaredMethod (System.Type type, System.String name, System.Type[] parameters, System.Type[] generics) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.PatchTools.GetOriginalMethod (HarmonyLib.HarmonyMethod attr) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
Rethrow as HarmonyException: Ambiguous match for HarmonyMethod[(class=SE_Rested, methodname=CalculateComfortLevel, type=Normal, args=undefined)]
HarmonyLib.PatchClassProcessor.ReportException (System.Exception exception, System.Reflection.MethodBase original) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.PatchClassProcessor.Patch () (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.Harmony.<PatchAll>b__11_0 (System.Type type) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.CollectionExtensions.Do[T] (System.Collections.Generic.IEnumerable`1[T] sequence, System.Action`1[T] action) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.Harmony.PatchAll (System.Reflection.Assembly assembly) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
HarmonyLib.Harmony.CreateAndPatchAll (System.Reflection.Assembly assembly, System.String harmonyInstanceId) (at <474744d65d8e460fa08cd5fd82b5d65f>:0)
EpicLoot.EpicLoot.Awake () (at <967dff30df8e47cf958595b66efa7e1c>:0)
UnityEngine.GameObject:AddComponent(Type)
BepInEx.Bootstrap.Chainloader:Start()
UnityEngine.Application:.cctor()

[Error  : Unity Log] MissingFieldException: Field 'ItemDrop/ItemData/SharedData.m_holdAnimationState' not found.
Stack trace:
EpicLoot.Crafting.CraftingItemExtensions.GetRunestoneRarity (ItemDrop+ItemData item) (at <967dff30df8e47cf958595b66efa7e1c>:0)
EpicLoot.ItemDataExtensions.GetRarity (ItemDrop+ItemData itemData) (at <967dff30df8e47cf958595b66efa7e1c>:0)
EpicLoot.EpicLoot.TryRegisterItems () (at <967dff30df8e47cf958595b66efa7e1c>:0)
EpicLoot.ObjectDB_CopyOtherDB_Patch.Postfix () (at <967dff30df8e47cf958595b66efa7e1c>:0)
(wrapper dynamic-method) ObjectDB.DMD<ObjectDB::CopyOtherDB>(ObjectDB,ObjectDB)
FejdStartup.SetupObjectDB () (at <035307060cbb4b30b916cd82ebd80490>:0)
(wrapper dynamic-method) FejdStartup.DMD<FejdStartup::Start>(FejdStartup)

[Info   : Unity Log] 12/10/2022 08:18:14: Ticket is System.Byte[] of length 369

[Info   : Unity Log] 12/10/2022 08:18:14: UI Group status changed Menu = True

[Info   : Unity Log] 12/10/2022 08:18:14: PlayFab logged in as "Steam_76561198832847702"

[Info   : Unity Log] 12/10/2022 08:18:14: PlayFab local entity ID is 55284A1513C0E36F

[Info   : Unity Log] 12/10/2022 08:18:16: UI Group status changed Menu = True

**Comment by javierb828** [comment]
> [Info : BepInEx] Loading [Extended Item Data Framework 1.0.7] [Warning: HarmonyX] AccessTools.DeclaredMethod: Could not find method for type Inventory and name AddItem and parameters (string, int, float, Vector2i, bool, int, int, long, string) [Error : Unity Log] ArgumentException: Undefined target method for patch method static bool ExtendedItemDataFramework.Inventory_AddItemFromLoad_Patch::Prefix(Inventory __instance, Boolean& __result, string name, int stack, float durability, Vector2i pos, bool equiped, int quality, int variant, long crafterID, string crafterName) Stack trace: HarmonyLib.PatchClassProcessor.PatchWithAttributes (System.Reflection.MethodBase& lastOriginal) (at <474744d65d8e460fa08cd5fd82b5d65f>:0) HarmonyLib.PatchClassProcessor.Patch () (at <474744d65d8e460fa08cd5fd82b5d65f>:0) Rethrow as HarmonyException: Patching exception in method null HarmonyLib.PatchClassProcessor.ReportException (System.Exception exception, System.Reflection.MethodBase original) (at <474744d65d8e460fa08cd5fd82b5d65f>:0) HarmonyLib.PatchClassProcessor.Patch () (at <474744d65d8e460fa08cd5fd82b5d65f>:0) HarmonyLib.Harmony.b__11_0 (System.Type type) (at <474744d65d8e460fa08cd5fd82b5d65f>:0) HarmonyLib.CollectionExtensions.Do[T] (System.Collections.Generic.IEnumerable`1[T] sequence, System.Action`1[T] action) (at <474744d65d8e460fa08cd5fd82b5d65f>:0) HarmonyLib.Harmony.PatchAll (System.Reflection.Assembly assembly) (at <474744d65d8e460fa08cd5fd82b5d65f>:0) HarmonyLib.Harmony.CreateAndPatchAll (System.Reflection.Assembly assembly, System.String harmonyInstanceId) (at <474744d65d8e460fa08cd5fd82b5d65f>:0) ExtendedItemDataFramework.ExtendedItemDataFramework.Awake () (at :0) UnityEngine.GameObject:AddComponent(Type) BepInEx.Bootstrap.Chainloader:Start() UnityEngine.Application:.cctor()
> 
> [Info : BepInEx] Loading [Epic Loot 0.8.9] [Error : Unity Log] AmbiguousMatchException: Ambiguous match found. Stack trace: System.RuntimeType.GetMethodImpl (System.String name, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Reflection.CallingConventions callConv, System.Type[] types, System.Reflection.ParameterModifier[] modifiers) (at <695d1cc93cca45069c528c15c9fdd749>:0) System.Type.GetMethod (System.String name, System.Reflection.BindingFlags bindingAttr) (at <695d1cc93cca45069c528c15c9fdd749>:0) HarmonyLib.AccessTools.DeclaredMethod (System.Type type, System.String name, System.Type[] parameters, System.Type[] generics) (at <474744d65d8e460fa08cd5fd82b5d65f>:0) HarmonyLib.PatchTools.GetOriginalMethod (HarmonyLib.HarmonyMethod attr) (at <474744d65d8e460fa08cd5fd82b5d65f>:0) Rethrow as HarmonyException: Ambiguous match for HarmonyMethod[(class=SE_Rested, methodname=CalculateComfortLevel, type=Normal, args=undefined)] HarmonyLib.PatchClassProcessor.ReportException (System.Exception exception, System.Reflection.MethodBase original) (at <474744d65d8e460fa08cd5fd82b5d65f>:0) HarmonyLib.PatchClassProcessor.Patch () (at <474744d65d8e460fa08cd5fd82b5d65f>:0) HarmonyLib.Harmony.b__11_0 (System.Type type) (at <474744d65d8e460fa08cd5fd82b5d65f>:0) HarmonyLib.CollectionExtensions.Do[T] (System.Collections.Generic.IEnumerable`1[T] sequence, System.Action`1[T] action) (at <474744d65d8e460fa08cd5fd82b5d65f>:0) HarmonyLib.Harmony.PatchAll (System.Reflection.Assembly assembly) (at <474744d65d8e460fa08cd5fd82b5d65f>:0) HarmonyLib.Harmony.CreateAndPatchAll (System.Reflection.Assembly assembly, System.String harmonyInstanceId) (at <474744d65d8e460fa08cd5fd82b5d65f>:0) EpicLoot.EpicLoot.Awake () (at <967dff30df8e47cf958595b66efa7e1c>:0) UnityEngine.GameObject:AddComponent(Type) BepInEx.Bootstrap.Chainloader:Start() UnityEngine.Application:.cctor()
> 
> [Error : Unity Log] MissingFieldException: Field 'ItemDrop/ItemData/SharedData.m_holdAnimationState' not found. Stack trace: EpicLoot.Crafting.CraftingItemExtensions.GetRunestoneRarity (ItemDrop+ItemData item) (at <967dff30df8e47cf958595b66efa7e1c>:0) EpicLoot.ItemDataExtensions.GetRarity (ItemDrop+ItemData itemData) (at <967dff30df8e47cf958595b66efa7e1c>:0) EpicLoot.EpicLoot.TryRegisterItems () (at <967dff30df8e47cf958595b66efa7e1c>:0) EpicLoot.ObjectDB_CopyOtherDB_Patch.Postfix () (at <967dff30df8e47cf958595b66efa7e1c>:0) (wrapper dynamic-method) ObjectDB.DMDObjectDB::CopyOtherDB(ObjectDB,ObjectDB) FejdStartup.SetupObjectDB () (at <035307060cbb4b30b916cd82ebd80490>:0) (wrapper dynamic-method) FejdStartup.DMDFejdStartup::Start(FejdStartup)

I am having the exact same error with EIDF 1.0.7.  Thunderstore says it has version 1.0.8 but when I download it it still gives 1.0.7.  As a result EPIC LOOT won't function properly in game.

If possible can we get a link to the most updated working versions of both EIDF and EPIC LOOT.  They are amazing mods and I can't wait to see what else you come up with.  Thanks a bunch for your help and all your hard work.

**Comment by TheGideon** [comment]
> I did find the Mistlands test versions of Both EIDF and EPIC LOOT. Dlad to say they appear to be working great. Thanks guys, you're awesome.

Where did u find them?

**Comment by maddog3521** [comment]
if he has a beta out then ill wait till he update r2modman with the new version

**Comment by javierb828** [comment]
I pulled them on GitHub. Can’t remember which of your threads I found them on. Now that you have the full release I have updated to them and still working wonderfully. Awesome work on them. JavierOn Dec 11, 2022, at 04:55, TheGideon ***@***.***> wrote:﻿

I did find the Mistlands test versions of Both EIDF and EPIC LOOT. Dlad to say they appear to be working great. Thanks guys, you're awesome.

Where did u find them?
